From 8157fe1fecd6696a534fef89bc4a1c12ff8f2818 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kristian=20H=C3=B8gsberg?= <krh@bitplanet.net>
Date: Wed, 11 Sep 2013 11:15:18 -0700
Subject: [PATCH 22/39] xwayland: Remove Xdnd selection watching code

We don't need this in the server, the wm can manage dnd proxying.
---
 hw/xfree86/xwayland/xwayland.c | 32 --------------------------------
 1 file changed, 32 deletions(-)

diff --git a/hw/xfree86/xwayland/xwayland.c b/hw/xfree86/xwayland/xwayland.c
index 36f7cbf..e8c58fd 100644
--- a/hw/xfree86/xwayland/xwayland.c
+++ b/hw/xfree86/xwayland/xwayland.c
@@ -54,7 +54,6 @@
  */
 
 static DevPrivateKeyRec xwl_screen_private_key;
-static Atom xdnd_atom;
 
 static void
 xserver_client(void *data, struct xserver *xserver, int fd)
@@ -178,31 +177,6 @@ xwl_screen_get(ScreenPtr screen)
     return dixLookupPrivate(&screen->devPrivates, &xwl_screen_private_key);
 }
 
-static void
-xwayland_selection_callback(CallbackListPtr *callbacks,
-			    pointer data, pointer args)
-{
-    SelectionInfoRec *info = (SelectionInfoRec *) args;
-    Selection *selection = info->selection;
-
-    switch (info->kind) {
-    case SelectionSetOwner:
-	if (selection->selection == xdnd_atom) {
-	    if (selection->window != None)
-		ErrorF("client %p starts dnd\n", info->client);
-	    else
-		ErrorF("client %p stops dnd\n", info->client);
-	}
-	break;
-    case SelectionWindowDestroy:
-	ErrorF("selection window destroy\n");
-	break;
-    case SelectionClientClose:
-	ErrorF("selection client close\n");
-	break;
-    }
-}
-
 struct xwl_screen *
 xwl_screen_create(void)
 {
@@ -231,12 +205,6 @@ xwl_screen_pre_init(ScrnInfoPtr scrninfo, struct xwl_screen *xwl_screen,
 
     noScreenSaverExtension = TRUE;
 
-    xdnd_atom = MakeAtom("XdndSelection", 13, 1);
-    if (!AddCallback(&SelectionCallback,
-		     xwayland_selection_callback, xwl_screen)) {
-	return FALSE;
-    }
-
     xorg_list_init(&xwl_screen->output_list);
     xorg_list_init(&xwl_screen->seat_list);
     xorg_list_init(&xwl_screen->damage_window_list);
-- 
1.8.3.1

